### PR TITLE
Document tophat in the gallery

### LINKED ISSUE #3580: Document top hat functions.
## Description

Assigned to myself.

Document (during the next sprint?)
* http://scikit-image.org/docs/stable/api/skimage.morphology.html#white-tophat
* http://scikit-image.org/docs/stable/api/skimage.morphology.html#black-tophat

- [ ] Add wikipedia page as ref : https://en.wikipedia.org/wiki/Top-hat_transform
- [ ] Add See also sections
- [ ] Add gallery example

I'm sure this is fairly unknown because not well documented...

### PR DESCRIPTION
**Pull request by sciunto**
## Description

Fixes #3580 

Note that something exists in http://scikit-image.org/docs/stable/auto_examples/xx_applications/plot_morphology.html#sphx-glr-auto-examples-xx-applications-plot-morphology-py 
With @emmanuelle We are convinced that this is quite hidden and redundancy is welcome.

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](https://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: doc/examples/filters/plot_tophat.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+"""
+=======
+Top hat
```
Comment (emmanuelle) [inline doc/examples/filters/plot_tophat.py]: Can you write a more explicit title like "removing small bright objects with a top hat"?

File: doc/examples/filters/plot_tophat.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+"""
+============================================
+Removing small objects with a top hat filter
+============================================
+
```
Comment (emmanuelle) [inline doc/examples/filters/plot_tophat.py]:
```suggestion

This example shows how to remove small objects from grayscale images. The top-hat transform is an operation that extracts small elements and details from given images. Here we use a white top-hat transform, which is defined as the difference between the input image and its (mathematical morphology) opening.
```

File: doc/examples/filters/plot_tophat.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+"""
+============================================
+Removing small objects with a top hat filter
```
Comment (emmanuelle) [inline doc/examples/filters/plot_tophat.py]:
```suggestion
Removing small objects in grayscale images with a top hat filter
```

**Comment by emmanuelle** [comment]
This one is not WIP any more, we should get it merged. @sciunto is it intentional that you don't accept push commits from other maintainers (no worries if it is)?

File: doc/examples/filters/plot_tophat.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+"""
+============================================
```
Comment (sciunto) [inline doc/examples/filters/plot_tophat.py]:
```suggestion
================================================================
```

**Comment by emmanuelle** [comment]
But you're very fast at accepting suggestions ;-)

File: doc/examples/filters/plot_tophat.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+"""
+================================================================
+Removing small objects in grayscale images with a top hat filter
+============================================
```
Comment (sciunto) [inline doc/examples/filters/plot_tophat.py]:
```suggestion
================================================================
```

**Comment by sciunto** [comment]
no, it's not intentional at all!

**Comment by sciunto** [comment]
but the box is ticked. Is there any other setting to adjust?  
b

**Comment by emmanuelle** [comment]
> but the box is ticked. Is there any other setting to adjust?
> b

I don't know...

File: doc/examples/filters/plot_tophat.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
+"""
+================================================================
+Removing small objects in grayscale images with a top hat filter
+================================================================
+
+This example shows how to remove small objects from grayscale images. The top-hat transform [1]_ is an operation that extracts small elements and details from given images. Here we use a white top-hat transform, which is defined as the difference between the input image and its (mathematical morphology) opening.
```
Comment (jni) [inline doc/examples/filters/plot_tophat.py]: This should be wrapped at 80c wide

File: doc/examples/filters/plot_tophat.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
+"""
+================================================================
+Removing small objects in grayscale images with a top hat filter
```
Comment (jni) [inline doc/examples/filters/plot_tophat.py]: small -> large? Based on the wikipedia example?

Comment (emmanuelle) [reply doc/examples/filters/plot_tophat.py]: the example shows how to keep large object only by removing the top-hat image. I don't think this is important...

File: doc/examples/filters/plot_tophat.py (hunk 6 above)
Comment (emmanuelle) [reply doc/examples/filters/plot_tophat.py]: oh yeah, I wrote this line in the suggestion editor which is very convenient but quite rustic...

**Comment by hmaarrfk** [comment]
Aren't weworking to deprecate these...

**Comment by sciunto** [comment]
They are different. With a tophat functions doing true tophat calculations and tophat functions in rank which were incorrect ;)

**Comment by hmaarrfk** [comment]
Ah thanks!

**Pull request merged**